bin/main.ml

```diff
@@ -1,18 +1,58 @@
 open Diffcessible
 open Cmdliner
+open ExtUnix.Specific
+
+let create_empty_diff_content () =
+  "diff --git a/Error b/Error\n--- a/Error\n+++ b/Error\n"
+
+let with_input_fd file_path f =
+  match file_path with
+  | Some path ->
+      if not (Sys.file_exists path) then failwith ("File not found: " ^ path);
+      In_channel.with_open_bin path f
+  | None -> f In_channel.stdin
+
+let with_setup_term f =
+  let tty_path = ttyname Unix.stdout in
+  let tty_fd = Unix.openfile tty_path [ Unix.O_RDWR ] 0o600 in
+  let term = Notty_unix.Term.create ~input:tty_fd ~output:tty_fd () in
+  let is_tty = Unix.isatty Unix.stdin in
+  Fun.protect
+    (fun () -> f is_tty term)
+    ~finally:(fun () ->
+      Notty_unix.Term.release term;
+      Unix.close tty_fd)
+
+let read_input ic =
+  try In_channel.input_all ic with _ -> create_empty_diff_content ()
 
 let main file_path =
-  let file = In_channel.open_bin file_path in
-  let s = In_channel.input_all file in
-  let patch = Patch.to_diffs s in
-  Interactive_viewer.start patch
+  with_setup_term (fun is_tty term ->
+      let input_content =
+        if is_tty then
+          match file_path with
+          | Some _ -> with_input_fd file_path read_input
+          | None -> create_empty_diff_content ()
+        else
+          Fun.protect
+            (fun () -> read_input In_channel.stdin)
```

Comment:
This branch do not match on `file_path`, which might ignore the file passed as an argument.